### PR TITLE
add project extra links

### PR DESCRIPTION
**Pull request by kalidiagne**
Show the project extra links section on the project details page.

**Desktop** 

Screenshot 2023-09-01 at 16 39 29

**Desktop with hover on link** 

Screenshot 2023-09-01 at 16 40 41

**Mobile**

Screenshot 2023-09-01 at 16 40 28

**JSON structure example** 

```js 
extraLinks: {
    research: [
      {
        label: "Link to get started",
        url: "https://google.it",
      },
    ],
    buildWith: [
      {
        label: "Link to get started",
        url: "https://google.it",
      },
    ],
    play: [
      {
        label: "Link to get started",
        url: "https://google.it",
      },
    ],
    learn: [
      {
        label: "Full technical specifications",
        url: "https://rate-limiting-nullifier.github.io/rln-docs/",
      },
      {
        label: "Another article related to the project",
        url: "https://google.it",
      },
      {
        label: "Presentation slides",
        url: "https://google.it",
      },
    ],
  }
```

### CODE REVIEW

**Review by ChialiT — COMMENTED** [review]
@kalidiagne I saw your screenshot, it seems working but does this take into consideration of the project lib?

for ex: if a project has "build" and "play" data, when user using the filter of either build or play, the project will show up in the filter result?

**Comment by kalidiagne** [comment]
> @kalidiagne I saw your screenshot, it seems working but does this take into consideration of the project lib?
> 
> for ex: if a project has "build" and "play" data, when user using the filter of either build or play, the project will show up in the filter result?

In your example the project will show by using the filters only when `build` and `play` are present in the `themes` part. 
To show the links we are filling the `extraLinks` object. 

having all in one can add more complexity to the structure 

we need to have a match of `themes` + `play` in `tags` & `extraLinks` as this example 

```js 
{
  "themes": [
    "build",
    "play"
  ],
  "extraLinks": {
    "build": [
      {
        "label": "Link to get started",
        "url": "https://google.it\""
      }
    ],
    "play": [
      {
        "label": "Link to get started",
        "url": "https://google.it\""
      }
    ]
  }
}
```

**Comment by ChialiT** [comment]
@kalidiagne got it. I totally forgot there is "theme" in the project.ts.
if in the instruction, we should let the project lead knows when they fill up these, they should make sure if there is extra link for "play", then the theme should include play as well, right?

**Comment by kalidiagne** [comment]
> @kalidiagne got it. I totally forgot there is "theme" in the project.ts. if in the instruction, we should let the project lead knows when they fill up these, they should make sure if there is extra link for "play", then the theme should include play as well, right?

@ChialiT 
Exactly, is not mandatory tho, but better to have the same.
I will add a file will *README* file with examples

**Review by thebeyondr — APPROVED** [review]

**Comment by kalidiagne** [comment]
@ChialiT added `README` with little description and examples

**Comment by ChialiT** [comment]
This is almost ready to go, missing the "project status", and let's change theme selected to "theme"

**Comment by kalidiagne** [comment]
> This is almost ready to go, missing the "project status", and let's change theme selected to "theme"

Here `project status`: https://github.com/privacy-scaling-explorations/website-v2/pull/83

File: components/project/project-extra-links.tsx [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+const ExtraLinkLabelMapping: Record<
+  ProjectExtraLinkType,
+  {
+    label: string
+    icon?: any
+  }
+> = {
+  buildWith: {
+    label: "Build With",
```
[the remark this answers is not in the document]
Comment (ChialiT) [reply components/project/project-extra-links.tsx:20]: @kalidiagne the label should be "Build with this tool"

**Review by ChialiT — COMMENTED** [review]
Need to update a label, otherwise its all good

**Review by ChialiT — APPROVED** [review]

**Pull request merged**